### PR TITLE
src/gnss.rs: add a gnss reader

### PR DESCRIPTION
**Pull request by nytowl**
The GNSS devices con't use the serial driver because they don't implement
the serial ioctls so the fail on open.

This driver should auto discover /dev/gnssX devices.

### CODE REVIEW

File: src/main.rs
Comment (zeenix) [inline src/main.rs:111]: Wouldn't having this seperate/independent from the RS232 above, mean we'll **always** get the `Failed to open serial device` from above match block for GNSS devices? Perhaps i'd be better to do GNSS detection if the RS232 code fails and error is not `NotFound`?

**Review by zeenix — CHANGES_REQUESTED** [review]

File: src/main.rs
Comment (nytowl) [reply src/main.rs:111]: This is my first attempts at rust so I think i understand how I might do that. I'll update the PR

Comment (nytowl) [reply src/main.rs]: I updated the code to check all devices before printing the error.

File: src/main.rs
Comment (zeenix) [inline src/main.rs:104]: nice! However, this makes code a bit hard to follow (due to indentation). Also, it'll get even harder to read, when/if someone adds support for another device type. So I'd suggest separating out the discovery part from RS232 module first (in a seperate  patch) and then adding GNSS support. That way this code will not have to change for addition of new devices at all and will be much more readable.

I'm sorry I didn't think of this before but as I mentioned, I've not looked at this for a while. If you don't have the time to do this, I understand but please do add a FIXME comment here then, how to improve this code (basically what I wrote above).

Comment (zeenix) [reply src/main.rs:104]: Oh and keep in mind that when I wrote this project, I was completely new to Rust. :)

Comment (nytowl) [reply src/main.rs:104]: Ok I'll add a FIXME as I'm short on time and skill with rust.

File: src/main.rs
Comment (zeenix) [inline src/main.rs:101]: In Rust, you want to use `//` comments.

```suggestion
    // FIXME: the discovery part should be separated from the RS232 module so that adding
    // more devices doesn't get even more convoluted.
```

**Review by zeenix — CHANGES_REQUESTED** [review]
LGTM otherwise.

File: src/main.rs
Comment (Kareema0) [reply src/main.rs:101]: I'll try to take care of the FIXME after this PR is merged (with the suggested change above). I'll continue from there with a new PR. I'm just learning Rust, but this seems to be an easy enough exercise to get some coding experience in Rust :-)

Comment (zeenix) [reply src/main.rs:101]: @Kareema0 that'd be great. As I say to every single person new to Rust, please do read [this blog post](https://ferrous-systems.com/blog/mindsets-and-expectations/).

**Review by zeenix — APPROVED** [review]

**Comment by nytowl** [comment]
This looks like everyone has approved. When does it get merged ?

**Comment by zeenix** [comment]
> This looks like everyone has approved. When does it get merged ?

Sorry, not sure why didn't press merge button.

**Pull request merged**
